### DISCUSSION TITLE
Axios request in middleware for next 13

### DISCUSSION THREAD

**Discussion by SySagar** [vercel/next.js#56322] [Help]
### Summary

I want to protect few routes in next js project. I am using the pp directory so all my folder are individual routes. I am thinking of adding a middleware.ts file at root where I have written the logic for JWT token validation from node js backend. Now it shows the error like this:

![Capture](https://github.com/vercel/next.js/assets/93787385/4c18f053-74e7-4958-a9a1-0817c2705965)

Can you help me in setting up the proper protected routes using nodejs middleware . I only want the verify request to go from the middleware of next js. Thanks! 

P.s. I have attached my middleware logic below.

### Additional information

```js
import { NextResponse } from 'next/server';
import { APIMethods } from './app/lib/axios/api';

export default function middleware() {

    const verifyToken = async () => {
        console.log("verifyToken");
        try {
          const response : any  = await APIMethods.user.verify();
            console.log("response",response);
          return response.status;
        } catch (error) {
          console.error("Error verifying token:", error);
          return null;  
        }
      };

  const userIsAuthenticated = verifyToken().then((data) => {
    if (data) {
      return true;
    }
    return false;
  });

  if (!userIsAuthenticated) {
    return NextResponse.redirect('/auth/login');
  }

  return NextResponse.next();
}
```

**Comment by jarrodmjack** [comment]
For anyone reading this in the future

It seems to be an issue specifically using axios on the edge (middleware.tsx)

If you want the requests to work, you should use fetch

If there is another way I am unsure of, I am happy to be corrected but through my testing it seems fetch is the only solution that works

**Comment by suhaotian** [comment]
Axios not work in `middleware.js`,  only fetch support, if you want similiar axios API, Use [xior](https://github.com/suhaotian/xior)

**Reply by fajarmaulanaaa** [reply]
can xior run on middleware?

**Reply by suhaotian** [reply]
Yes, definitely, I build this one of reason is axios not work with middleware.js
